service/src/main/kotlin/io/provenance/digitalcurrency/consortium/stream/Events.kt

```diff
@@ -179,3 +182,28 @@ private fun StreamEvent.toTransfer(): Transfer =
         height = height,
         txHash = txHash
     )
+
+fun EventBatch.migrations(contractAddress: String): Migrations =
+    events
+        .filter { event ->
+            val contractAddressAttr = event.getAttribute(ATTRIBUTE_CONTRACT_ADDRESS_CURRENT)
+            event.eventType == MIGRATE_EVENT &&
+                contractAddress == contractAddressAttr
+        }.map { event -> event.toMigration() }
+
+typealias Migrations = List<Migration>
+
+data class Migration(
+    val contractAddress: String,
+    val codeId: String,
+    val height: Long,
+    val txHash: String
+)
+
+private fun StreamEvent.toMigration(): Migration =
+    Migration(
+        contractAddress = getAttribute(ATTRIBUTE_CONTRACT_ADDRESS),
```

Comment:
is this attribute going to exist when we filter by prefix _? should everything be looking at prefix _ now? or should we find first one in list of if [contract_address, _contract_address]?

Reply:
I need to test the other event types before we do a release to main. For the migration event, only `_contract_address` was present.

Reply:
so is this line using the correct attribute?

Reply:
good catch. it isn't used anywhere - removing. Unless we will have > 1 contract some day and I should persist it...